corehq/form_processor/submission_post.py

```diff
@@ -98,6 +105,10 @@ def __init__(self, instance=None, attachments=None, auth_context=None,
         self.is_openrosa_version3 = self.openrosa_headers.get(OPENROSA_VERSION_HEADER, '') == OPENROSA_VERSION_3
         self.track_load = form_load_counter("form_submission", domain)
 
+        self.pre_processing_steps = [
+            to_function(class_name)() for class_name in settings.XFORM_PRE_PROCESSORS.get(self.domain, [])
```

Comment:
what do you think about caching the conversion from string to class to avoid having to do that on every request?

Reply:
That is definitely possible. The one reason I avoided that was to avoid the risk of passing context from one request to another and start afresh. Even though the current pre processing steps won't run into such a scenario but if more are added, it might cause unexpected issues. I am happy to switch to caching this if you think that is not really needed.

Reply:
I don't mean caching the class instances, just the classes themselves, ie `to_function(class_name)`